unittests/compiler/subsortmap.cpp

```diff
@@ -0,0 +1,50 @@
+#include <boost/test/unit_test.hpp>
+#include <kllvm/ast/AST.h>
+
+using namespace kllvm;
+
+BOOST_AUTO_TEST_SUITE(SubsortMapTest)
+
+BOOST_AUTO_TEST_CASE(insert_contains) {
+  SortSet sortSet;
+  SymbolSet symbolSet;
+  auto subsortMap = SubsortMap{};
+
+  auto sort = kore_sort_variable::create("Sort");
+  auto subsort = kore_sort_variable::create("Subsort 1");
+  auto subsort2 = kore_sort_variable::create("Subsort 2");
+
+  subsortMap[sort.get()].insert(subsort.get());
+  subsortMap[sort.get()].insert(subsort2.get());
+
+
+  BOOST_CHECK(subsortMap[sort.get()].contains(subsort.get()));
+  BOOST_CHECK(subsortMap[sort.get()].contains(subsort2.get()));
+  BOOST_CHECK(subsortMap[sort.get()].size() == 2);
+  BOOST_CHECK(subsortMap.size() == 1);
+  BOOST_CHECK(!subsortMap[sort.get()].contains(sort.get()));
+  BOOST_CHECK(!subsortMap[sort.get()].empty());
+}
+
+BOOST_AUTO_TEST_CASE(subsorts_supersorts) {
+  auto subsort1 = kore_sort_variable::create("bar");
+  auto subsort2 = kore_sort_variable::create("baz");
```

Comment:
It would be good to check that this works for concrete sorts; the underlying mechanism is the same and so it should work similarly, but asserting over concrete sorts will be closer to what's actually happening in the compiler when we deal with real KORE.

Reply:
How can I do that? Could you point me to the right function, please?

Reply:
Just create a `kore_composite_sort` rather than a `kore_sort_variable`

Reply:
Done!